### DISCUSSION TITLE
Regression in OpenAPI related checks during `app.include_router(router)`

### DISCUSSION THREAD

**Discussion by Kyle-sandeman-mrdfood** [fastapi/fastapi#8380] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Union

from fastapi import FastAPI, APIRouter
from fastapi.responses import FileResponse
from pydantic import BaseModel


class Item(BaseModel):
    id: str
    value: str


responses = {
    404: {"description": "Item not found"},
    302: {"description": "The item was moved"},
    403: {"description": "Not enough privileges"},
    "5XX": {"description": "Server errors", "model": Item},
}


app = FastAPI(responses=responses)
router = APIRouter()

@router.get(
    "/items/{item_id}",
    response_model=Item,
)
async def read_item(item_id: str, img: Union[bool, None] = None):
    if img:
        return FileResponse("image.png", media_type="image/png")
    else:
        return {"id": "foo", "value": "there goes my hero"}

app.include_router(router)
```

### Description

In approx 0.76.0 a regression was observed related to the OpenAPI responses no longer accepting patterns of the form 4XX 5XX etc.
```
>>> app.include_router(router)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/venv/lib/python3.8/site-packages/fastapi/applications.py", line 420, in include_router
    self.router.include_router(
  File "/tmp/venv/lib/python3.8/site-packages/fastapi/routing.py", line 738, in include_router
    self.add_api_route(
  File "/tmp/venv/lib/python3.8/site-packages/fastapi/routing.py", line 565, in add_api_route
    route = route_class(
  File "/tmp/venv/lib/python3.8/site-packages/fastapi/routing.py", line 422, in __init__
    assert is_body_allowed_for_status_code(
  File "/tmp/venv/lib/python3.8/site-packages/fastapi/utils.py", line 24, in is_body_allowed_for_status_code
    current_status_code = int(status_code)
ValueError: invalid literal for int() with base 10: '5XX'
```
Please refer to this PR https://github.com/tiangolo/fastapi/pull/5145 which introduced the new code

### Operating System

Linux

### FastAPI Version

0.81.0

### Python Version

3.8.13

### Additional Context

I can't actually commit to a PR/monitoring this repo/such. I will likely sponsor the project soon as I love it, but I simply don't have that kind of time

**Comment by JarroVGIT** [comment, marked as answer]
A fix is ready here https://github.com/tiangolo/fastapi/pull/5187

**Comment by Kyle-sandeman-mrdfood** [comment]
Thanks, now just a question of who can approve+merge it
